ui/app/pages/send/send-content/add-recipient/ens-input.js

```diff
@@ -0,0 +1,274 @@
+            onChange={this.onChange}
+            value={selectedAddress || input}
+          />
+          <div
+            className={c('ens-input__wrapper__action-icon', {
+              'ens-input__wrapper__action-icon--erase': input,
+              'ens-input__wrapper__action-icon--qrcode': !input,
+            })}
+            onClick={() => {
+              if (input) {
+                this.resetInput()
+              } else {
+                this.props.scanQrCode()
+              }
+            }}
+          />
+        </div>
+      </div>
+    )
+  }
+
+  renderSelected () {
+    const { t } = this.context
+    const { className, selectedAddress, selectedName, addressBook } = this.props
+    const contact = addressBook.filter(item => item.address === selectedAddress)[0] || {}
+    const name = contact.name || selectedName
+
+
+    return (
+      <div className={c('ens-input', className)}>
+        <div
+          className="ens-input__wrapper ens-input__wrapper--valid"
+        >
+          <div className="ens-input__wrapper__status-icon" />
+          <div
+            className="ens-input__wrapper__input ens-input__wrapper__input--selected"
+            placeholder={t('recipientAddress')}
+            onChange={this.onChange}
+          >
+            <div className="ens-input__selected-input__title">
+              {name || selectedAddress}
+            </div>
+            { name && <div className="ens-input__selected-input__subtitle">{selectedAddress}</div> }
+          </div>
+          <div
+            className="ens-input__wrapper__action-icon ens-input__wrapper__action-icon--erase"
+            onClick={this.resetInput}
+          />
+        </div>
+      </div>
+    )
+  }
+
+  ensIcon (recipient) {
+    const { hoverText } = this.state
+
+    return (
+      <span
+        className="#ensIcon"
+        title={hoverText}
+        style={{
+          position: 'absolute',
+          top: '16px',
+          left: '-25px',
+        }}
+      >
+        { this.ensIconContents(recipient) }
+      </span>
+    )
+  }
+
+  ensIconContents () {
+    const { loadingEns, ensFailure, ensResolution, toError } = this.state || { ensResolution: ZERO_ADDRESS }
+
+    if (toError) return
+
+    if (loadingEns) {
+      return (
+        <img
+          src="images/loading.svg"
+          style={{
+            width: '30px',
+            height: '30px',
+            transform: 'translateY(-6px)',
+          }}
+        />
+      )
+    }
+
+    if (ensFailure) {
+      return <i className="fa fa-warning fa-lg warning'" />
+    }
+
+    if (ensResolution && (ensResolution !== ZERO_ADDRESS)) {
+      return (
+        <i
+          className="fa fa-check-circle fa-lg cursor-pointer"
+          style={{ color: 'green' }}
+          onClick={event => {
+            event.preventDefault()
+            event.stopPropagation()
+            copyToClipboard(ensResolution)
+          }}
+        />
+      )
+    }
+  }
+}
+
+export default connect(
```

Comment:
In most places we follow the pattern of creating a `.container.js` file for connection to redux. Do you have a preference for doing this within the same file that defines the component?

Reply:
Fixed